apps/arweave/src/ar_mining_server.erl

```diff
@@ -732,49 +702,83 @@ prepare_solution(poa2, Candidate, Solution) ->
 	#mining_candidate{
 		chunk2 = Chunk2, h0 = H0, nonce = Nonce,
 		partition_upper_bound = PartitionUpperBound } = Candidate,
+
+	case prepare_poa(poa2, Candidate, CurrentPoA2) of
+		{ok, PoA2} ->
+			prepare_solution(poa1, Candidate, Solution#mining_solution{ poa2 = PoA2 });
+		{error, Error} ->
+			Chunk2Binary = case Chunk2 of
+				not_set ->
```

Comment:
It is not a realistic case that we do not have the second chunk and leave it to the exit peer to fetch it, does it?

Reply:
Yeah you're right, that case statement was preserving the existing logic but fixing a crash. the previous `prepare_solution(poa2` call didn't check whether Chunk2 was set or not and just sent it directly to the exit peer here: https://github.com/ArweaveTeam/arweave/pull/628/files#diff-ea8e957c63acb7497b0ed4f9201b40f18e5d5b4fa371e67b4cff05850e4df854L773

Unfortunately when Chunk2 was `not_set` it caused a crash during `ar_serialize`. So the above change just fixes that `ar_serialize` crash, but doesn't change the overall logic of deferring to the exit peer (which I agree, doesn't buy us much if it's the Chunk data that's missing rather than proof data)

That said I didn't fully investigate why `chunk2` was `not_set` in the first place... so unclear to me how we get to this code path anyways. The `case` statement "fix" was just in response to the stack traces shared by miners which were showing a crash in `ar_serialize` due to the poa.chunk value being `not_set` rather than a binary.

Reply:
Hm, interesting